### PR TITLE
Adding appveyor config

### PR DESCRIPTION
**Pull request by tyler-ball**
\cc @jdmundrawala @fnichol @schisamo

### CODE REVIEW

File: lib/kitchen/driver/ssh_base.rb [hunk 1]
```diff
@@ -106,6 +106,7 @@ def verify(state) # rubocop:disable Metrics/AbcSize
 
         instance.transport.connection(backcompat_merged_state(state)) do |conn|
           conn.execute(env_cmd(verifier.init_command))
+          require 'pry'; binding.pry
```
Comment (schisamo) [inline lib/kitchen/driver/ssh_base.rb]: @tyler-ball Did you mean to commit this?

**Comment by smurawski** [comment]
There are several main categories of problems with the tests on Windows:
* Anything that specified a path starting with '/' was resolved as 'C:/' - so lots of the /rooty stuff was failing to match.  I added a helper function to resolve and add the system drive for Windows.
* All the net ssh related tests included some chmod stuff, which is a NOOP on Windows, so I added some conditionals to vary the expected mode of files based on OS.
* There were a couple of tests checking if a file was executable, and since that means nothing in the context of the windows file system, I put a guard around those.

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+version: "master-{build}"
+
+os: Windows Server 2012
+platform:
+  - x64
+
+environment:
+  matrix:
+    #- ruby_version: "193"
```
Comment (jaym) [inline appveyor.yml]: delete it?

File: appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    - ruby_version: "200"
+    - ruby_version: "215"
+
+clone_folder: c:\projects\busser
```
Comment (jaym) [inline appveyor.yml]: `c:\projects\test_kitchen`?

File: appveyor.yml (hunk 2 above)
Comment (smurawski) [inline appveyor.yml]: @jaym the 193?

**Comment by jaym** [comment]
@fnichol @tyler-ball do you want to put this under the chef appveyor account or are you going to have a separate one?

File: appveyor.yml (hunk 2 above)
Comment (jaym) [inline appveyor.yml]: yea

File: appveyor.yml (hunk 3 above)
Comment (tyler-ball) [inline appveyor.yml]: copypasta error - good catch

**Comment by tyler-ball** [comment]
Hey @smurawski - thanks for working on this.  I added you as the assignee for tracking since you put in the last commit.  Feel free to unassign yourself if it causes issues for your normal workflow.

**Comment by smurawski** [comment]
I'm first going to verify that we can get the unit and quality tests to run.  I'm having a heck of a time getting cucumber to work on windows locally, but getting unit tests running is a good first step.

File: appveyor.yml (hunk 3 above)
Comment (smurawski) [inline appveyor.yml]: Will fix that in the next commit.  Forgot in my last.

File: appveyor.yml (hunk 2 above)
Comment (smurawski) [inline appveyor.yml]: Will do.

**Comment by smurawski** [comment]
@tyler-ball @fnichol If this all comes back green, I'd like to get this merged into master.  I know it won't have appveyor running the feature specs, but I'd like to get some coverage for testing on Windows in place while I figure out the cucumber story (looks like there are issues with ruby 2.1.x on Windows - plus some OS bitness issues).

**Comment by tyler-ball** [comment]
Thanks so much @smurawski

**Pull request merged**